source-braintree-native/source_braintree_native/api.py

```diff
@@ -93,6 +99,8 @@ async def fetch_transactions(
         log: Logger,
         log_cursor: LogCursor,
 ) -> AsyncGenerator[IncrementalResource | LogCursor, None]:
+    # Yield to the event loop to prevent starvation.
+    await asyncio.sleep(0)
```

Comment:
I agree, this isn't intended to try and let multiple calls progress concurrently. I won't be able to achieve that without making the API calls asynchronous. My main concern was that if a stream isn't caught up to the present, it would continue trying to get caught up by immediately reinvoking `fetch_foobar` & blocking any of the other streams from making API calls since there aren't any places within `fetch_foobar` that yields to the event loop. I also thought this might prevent the `SIGQUIT` mechanism from working if one of the streams was trying to catch up to the present. But I tested locally, and sending a `SIGQUIT` still works without these `await asyncio.sleep(0)` lines, so there's still an `await` in the CDK that's yielding to the event loop and preventing these issues (I think it's [here](https://github.com/estuary/connectors/blob/1c80e71768e0ec98443e2204b79dcccffe7688f6/estuary-cdk/estuary_cdk/capture/common.py#L655-L657)).

I'll update the PR to remove these `await asyncio.sleep(0)` lines since they aren't needed. I'll keep the error message fix and merge that.